src/libraries/Microsoft.Extensions.Logging.Console/src/ConsoleLoggerProcessor.cs

```diff
@@ -31,63 +73,153 @@ public ConsoleLoggerProcessor(IConsole console, IConsole errorConsole)
                 Name = "Console logger queue processing thread"
             };
             _outputThread.Start();
+            _processorId = ++s_processorCount;
+            s_processors.Add(this);
         }
 
-        public virtual void EnqueueMessage(LogMessageEntry message)
+        static ConsoleLoggerProcessor()
         {
-            if (!_messageQueue.IsAddingCompleted)
-            {
-                try
-                {
-                    _messageQueue.Add(message);
-                    return;
-                }
-                catch (InvalidOperationException) { }
-            }
+            s_meter.CreateObservableGauge<long>("queue-size", GetQueueSize);
```

Comment:
I just double checked and I see we have several other counters in other areas of .NET that use "queue-length" but none which use "queue-size". I'd recommend changing this to match our precedent elsewhere.
```suggestion
            s_meter.CreateObservableGauge<long>("queue-length", GetQueueSize);
```